### ISSUE TITLE
Create git tags corresponding to new upstream mastodon versions 

### ISSUE THREAD

**Issue by CSDUMMI** [glitch-soc/mastodon#2360]
### Pitch

Whenever a new release of mastodon is merged into glitch, glitch should publish a new tag.

The tag format maybe `v<mastodon-version>-glitch` or `v<mastodon-version>-glitch-<date>` to distinguish multiple tags created for the same mastodon version.

The most recent stable Mastodon release would thus result in the tag `v4.1.6-glitch` or `v4.1.6-glitch-2023-08-09`.

### Motivation

By using tags, glitch would make it vastly easier for instances to upgrade to new versions of Mastodon without including untested, unstable or insecure changes introduce into Mastodon and merged into glitch in-between releases.

**Comment by ClearlyClaire** [comment]
Hi! I understand the reasoning, but merging/porting stuff from upstream is a significant amount of work, work that can introduce unintended issues, so a stable upstream release does not mean there is no additional issue in glitch-soc nor that it has been tested as thoroughly.

**Comment by compuguy** [comment]
I agree with @nightpaws on having tags/builds for Mastodon releases. It would simplify the process of determining what build matches closest to an upstream Mastodon release. I get that this "tagged" release may or may not be stable, and that is the risk of using the glitch edition. So, I get @ClearlyClaire's concerns here as well.

**Comment by Plastikmensch** [comment]
Another issue with tagged releases for glitch-soc is that it would require backports and security fixes, adding additional load. These are usually included in upstream merges or simultaneously merged into glitch-soc.
It is always safer to run the latest main, which is usually ahead of any upstream release.

Switching between vanilla and glitch-soc is usually not a problem as long as migrations are run.
It's essentially the same as updating from one upstream release to another.
